src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy

```diff
@@ -328,4 +330,25 @@ class UnityPlugin implements Plugin<Project> {
             task.group = GROUP
         }
     }
+
+    private static void addAddUPMPackagesTask(Project project, final UnityPluginExtension extension) {
+        def manifestFile = project.layout.projectDirectory.file("Packages/manifest.json")
+        def addUPMPackagesTask = project.tasks.register(Tasks.addUPMPackages.toString(), AddUPMPackages) {task ->
+            task.group = GROUP
+            task.manifestPath.convention(manifestFile)
+            task.upmPackages.putAll(extension.upmPackages)
+            if (extension.enableTestCodeCoverage.getOrElse(false)) {
+                task.upmPackages.put("com.unity.testtools.codecoverage", "1.1.0")
+            }
```

Comment:
```suggestion
            task.upmPackages.put("com.unity.testtools.codecoverage", extension.enableTestCodeCoverage.map({
                it ? "1.1.0" : null
            }))
```

Can you test if this transformation does what it should?
Or maybe this way:
```
task.upmPackages.putAll(project.provider({
                def map = [:]
                if (extension.enableTestCodeCoverage.isPresent() && extension.enableTestCodeCoverage) {
                    map["com.unity.testtools.codecoverage"] = "1.1.0"
                }
                map
            }))
```

or like this:

```
task.upmPackages.putAll(extension.enableTestCodeCoverage.map({
                def map = [:]
                if (it) {
                    map["com.unity.testtools.codecoverage"] = "1.1.0"
                }
                map 
            }))
```

Mainly we should try not to call getOrElse to early. Again since we are in a configuration avoidance block that might all yield the same anyways.

Reply:
So, weird thing: the implementation on the suggested change does not work, as if the provider is empty, it empties the whole `upmPackages` provider (which makes no sense). Anyway, I used a variant of the last implementation and it worked just fine. However, it took me a while to discover this, thanks gradle.